spec/lib/generators/module_component/module_components_generator_spec.rb

```diff
@@ -0,0 +1,251 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'generators/module_component/module_component_generator'
+require 'generators/module/module_generator'
+
+RSpec.describe 'ModuleComponent', type: :generator do
+  before(:all) do
+    @original_stdout = $stdout
+    # Redirect stdout to suppress generator output
+    $stdout = File.open(File::NULL, 'w')
+  end
+
+  after(:all) do
+    $stdout = @original_stdout
+  end
+
+  describe 'creates a controller' do
+    before(:all) do
+      ModuleGenerator.new(['foo']).create_directory_structure
+    end
+
+    after(:all) { FileUtils.rm_rf(Dir[Rails.root.join('modules', 'foo')]) }
+
+    let(:path) { Rails.root.join('modules', 'foo', 'app', 'controllers') }
+
+    it 'creates the module controller file' do
+      module_generator = ModuleComponentGenerator.new(%w[foo controller])
+      module_generator.create_component
+      expect(module_generator.commit_message_methods).to eq(['controller'])
+      expect(File).to exist("#{path}/foo/v0/foo_controller.rb")
+    end
+  end
+
+  describe 'creates a serializer' do
+    before(:all) do
+      ModuleGenerator.new(['foo']).create_directory_structure
+    end
+
+    after(:all) { FileUtils.rm_rf(Dir[Rails.root.join('modules', 'foo')]) }
+
+    let(:path) { Rails.root.join('modules', 'foo', 'app', 'serializers') }
+
+    it 'creates the module serializer file' do
+      module_generator = ModuleComponentGenerator.new(%w[foo serializer])
+      module_generator.create_component
+      expect(module_generator.commit_message_methods).to eq(['serializer'])
+      expect(File).to exist("#{path}/foo/v0/foo_serializer.rb")
+    end
+  end
+
+  describe 'creates a model' do
+    before(:all) do
+      ModuleGenerator.new(['foo']).create_directory_structure
+    end
+
+    after(:all) { FileUtils.rm_rf(Dir[Rails.root.join('modules', 'foo')]) }
+
+    let(:path) { Rails.root.join('modules', 'foo', 'app', 'models') }
+
+    it 'creates the module model file' do
+      module_generator = ModuleComponentGenerator.new(%w[foo model])
+      module_generator.create_component
+      expect(module_generator.commit_message_methods).to eq(['model'])
+      expect(File).to exist("#{path}/foo/v0/foo_model.rb")
```

Comment:
Should this be foo.rb instead of foo_model.rb?

Reply:
✔️